### PR TITLE
Add support for sidecars

### LINKED ISSUE #37: Add support for sidecars
/assign @fenglixa

### PR DESCRIPTION
**Pull request by fenglixa**
Fixes #37
After the fixes, compiled sidecars tekton pipeline could be ran successful:
```
Name:           sidecar-run-lkg6w
Namespace:      default
Pipeline Ref:   sidecar

🌡️  Status

STARTED          DURATION   STATUS
14 minutes ago   1 minute   Succeeded

📦 Resources

 No resources

⚓ Params

 No params

🗂  Taskruns

 NAME                                 TASK NAME   STARTED          DURATION     STATUS
 ∙ sidecar-run-lkg6w-echo-7qcqc       echo        14 minutes ago   20 seconds   Succeeded
 ∙ sidecar-run-lkg6w-download-tf88f   download    14 minutes ago   36 seconds   Succeeded
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfp-tekton/42)

### CODE REVIEW

**Comment by ckadner** [comment]
Thanks for your PR @fenglixa. 

The `sidecar` use case looks good, but could please you separate out the second `with_param` use case into a separate PR? `LoopArguments` and `ParallelFor` should be tackled indedpendently.

File: sdk/python/tests/compiler/testdata/sidecar.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Google LLC
```
Comment (ckadner) [inline sdk/python/tests/compiler/testdata/sidecar.py]: `# Copyright 2020 kubeflow.org`

File: sdk/python/tests/compiler/testdata/sidecar.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+apiVersion: tekton.dev/v1beta1
```
Comment (ckadner) [inline sdk/python/tests/compiler/testdata/sidecar.yaml]: add the license header

**Comment by fenglixa** [comment]
@ckadner, Great thanks for your reivew comments.  I addressed them, please help check whether it's OK for your now?

**Comment by ckadner** [comment]
Thank you @fenglixa this is great!

**Comment by ckadner** [comment]
just for reference, this is the output from my pipeline run:

`tkn pipeline start sidecar --showlog`:
```
Pipelinerun started: sidecar-run-v8jt7
Waiting for logs to be available...
[download : download] Connecting to localhost:5678 (127.0.0.1:5678)
[download : download] saving to '/tekton/results/downloaded'
[download : download] downloaded           100% |********************************|    14  0:00:00 ETA
[download : download] '/tekton/results/downloaded' saved

[download : sidecar-echo] 2020/03/20 08:20:46 Server is listening on :5678
[download : sidecar-echo] 2020/03/20 08:20:58 localhost:5678 127.0.0.1:47932 "GET / HTTP/1.1" 200 14 "Wget" 13.619µs
[download : sidecar-echo] 2020/03/20 08:20:59 [ERR] Unknown signal terminated
[echo : echo] hello world
```

**Comment by fenglixa** [comment]
@ckadner,  good catch, acturally, I also noticed this output, but after investigated, it is an known issue of current implementation of tekton pipeline for sidecars,  which has no any function impacts. 
It was mentioned in the readme of tekton pipeline:
https://github.com/tektoncd/pipeline/blob/master/docs/developers/README.md
With the content of:
> There are known issues with the existing implementation of sidecars:
> - When the nop image does provide the sidecar's command, the sidecar will continue to run even after nop has been swapped into the sidecar container's image field. See the issue tracking this bug for the issue tracking this bug. Until this issue is resolved the best way to avoid it is to avoid overriding the nop image when deploying the tekton controller, or ensuring that the overridden nop image contains as few commands as possible.
> - kubectl get pods will show a Completed pod when a sidecar exits successfully but an Error when the sidecar exits with an error. This is only apparent when using kubectl to get the pods of a TaskRun, not when describing the Pod using kubectl describe pod ... nor when looking at the TaskRun, but can be quite confusing.

**Comment by fenglixa** [comment]
@animeshsingh, could you help review? I may have another PR which is based on this fixes. Thanks very much!

**Comment by animeshsingh** [comment]
captured the output comments in an issue to come back to
https://github.com/kubeflow/kfp-tekton/issues/44

**Comment by fenglixa** [comment]
Resolved the conflicts for compiler_tests.py

**Comment by fenglixa** [comment]
Thanks @animeshsingh

**Pull request merged**